src/Manager.ts

```diff
@@ -0,0 +1,180 @@
+import { words } from './constants/Wordlist';
+import { GameUtility as gu } from './GameUtility';
+import { RuleEnforcer as re } from './RuleEnforcer';
+import { Team, PlayerLocation } from './constants/Constants';
+
+import ws = require('ws');
+import { shuffle } from 'lodash'
+
+export class Manager {
+	playerStates: Map<ws, PlayerState>;
+	loners: Map<string, Player>;
+	lobbies: Map<string, Lobby>;
+	games: Map<string, Game>;
+	usedGids: Set<string>;
+	availableGids: string[];
+
+	// make instance of Game class
+	constructor() {
+		this.playerStates = new Map();
+		this.loners = new Map();
+		this.lobbies = new Map();
+		this.games = new Map();
+		this.usedGids = new Set();
+		this.availableGids = words.map(word => word.toLocaleLowerCase());
+	}
+
+	handleClose(socket: ws) {
+		if(!this.playerStates.has(socket)) {
+			console.log('User hadn\'t set name, no player to remove');
+			return;
+		}
+
+		// player exists somewhere
+		let state = this.playerStates.get(socket) as PlayerState;
+
+		switch(state.getLoc()) {
+			case PlayerLocation.loner:
+				this.removeLoner(socket, state.getPid())
+				console.log(`Removed player from loners with id: ${state.getPid()}`)
+				break;
+			case PlayerLocation.lobby:
+				if(this.lobbies.has(state.getGid())) {
+					let lobby = (this.lobbies.get(state.getGid()) as Lobby);
+					lobby.removeLoiterer(socket);
+					console.log(`Removed player from lobby ${state.getGid()} with id: ${state.getPid()}`)
+
+					if(lobby.empty()) {
+						this.lobbies.delete(state.getGid());
+						console.log(`Deleted lobby ${state.getGid()}`);
+					}
+				}
+				break;
+			case PlayerLocation.game:
+				if(this.games.has(state.getGid())) {
+					let game = (this.games.get(state.getGid()) as Game);
+					game.removeAgent(socket);
+					console.log(`Removed player from game ${state.getGid()} with id: ${state.getPid()}`);
+
+					if(game.empty()) {
+						this.games.delete(state.getGid());
+						console.log(`Deleted game ${state.getGid()}`);
+					}
+				}
+				break;
+			default:
+				console.log('Sorry, unknown player location.')
+		}
+	}
+	
+	handleMessage(message: any, socket: ws) {
+		console.log(message)
+
+		if(message.hasOwnProperty('action')) {
+			let action: string = message.action;
+			
+			switch(action) {
+				case "setName":
+					console.log('Case setName reached');
+					if (re.isValidName(message.name)) {
+						// this.game.registerLoiterer(message.name, socket);
+						this.registerPlayer(message.name, socket);
+					}
+					break;
+
+				case "createLobby":
+					if(this.availableGids.length <= 0)
+						throw new Error('If these are real games that\'s awesome, now add more gids')
+					
+					let tempGids = shuffle(this.availableGids)
+					let newGid: string = tempGids.pop() as string;
+					this.availableGids = tempGids;
+
+					this.lobbies.set(newGid, new Lobby(newGid));
+					this.placePlayerInLobby(message.pid, newGid, socket);
+					break;
+
+				case "joinLobby":
```

Comment:
This function should have error handling to let the front-end know if no lobby can be found.